### ISSUE TITLE
Rival 600 support?

### ISSUE THREAD

**Issue by RetroDevil** [antonpup/Aurora#1199]
### What are you experiencing an issue with?

- [ ] Aurora
- [ ] Aurora supported game, (Fill in the game)
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [x] Aurora supported device, (Fill in problematic devices)
- [ ] Aurora Scripts
- [ ] Aurora Updater
- [ ] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)

### What kind of an issue is it?

- [ ] Crash
- [ ] Performance problems
- [ ] Visual bug
- [ ] Localization issue
- [x ] Enhancement suggestion
- [ ] Device issue
- [ ] Other, (Fill in another type of issue here)

Anyway to add support for SteelSeries Rival 600 ? This mouse doesn't work on any other preffered mouse profiles

**Comment by simon-wh** [comment]
It should just work as the Steelseries integration doesn't care about the specific device so it should just light (at least some of the zones) when you have something like the Rival 300 selected.

**Comment by RetroDevil** [comment]
It still don't work. Even rival 300 profiles.

**Comment by FHeilmann** [comment]
The rival 600 has 8 zones (as opposed to the up to two currently supported by aurora). 
I guess one would have to look at the Steelseries SDK and find out how to address these new zones as I am sure they just have new identifiers. maybe @brainbug89 or @ToadKing can comment on this.

**Comment by FTWOblivion** [comment]
Bump because i just got the steel series 600 and was sure it was supported on here. I know it may take a little extra work but the Rival 600 is set to be the new top mouse on the block and im sure many people will be using it and hoping it has Aurora support, sorry to sound Needy but i do think we Need proper implementation here for the Rival 600, i wish i had the tech savvy knowledge to do it myself or help, but all i can really do is express and reiterate to the dev reading this that we do want it

**Comment by neo722** [comment]
i have the same issue, steelseries rival 600 not working at all with Aurora, i tried everything, thanks

**Comment by simon-wh** [comment]
In the code I can't find a reason why the Rival 600 wouldn't work, what is shown in SSE3 when you have Aurora running? For working devices IIRC it should show that the lighting for the zones have been taken over and there should be an integration for Aurora listed

**Comment by Gurjot95** [comment]
I remember @SnakePin said the Hidlibrary got broken again so he is looking for the fix in SS.

**Comment by SnakePin** [comment]
What Gurjot95 said is true and Rival 600 is 8-zone, devices with zones require different support from sdk but GameSense sdk doesn't support 8 zone devices yet. That's why I will be working on reverse engineering Rival 600 as soon as possible.

**Comment by SnakePin** [comment]
And the reason of hidlibrary not working anymore is related to latest Windows 10 updates. I will be working on finding replacement for the library or directly use Windows API in the code.

**Comment by ToadKing** [comment]
The Rival 600 should be usable through GameSense using the `rgb-8-zone` device type, but I'm noticing our public documentation hasn't been updated to reflect this. I'll bring this up internally to get this updated.

**Comment by simon-wh** [comment]
Can people with the Rival 600 try [this build](https://ci.appveyor.com/project/antonpup/aurora/build/0.6.2b-464/artifacts) to see if it gets some of the lighting working? I added the `rgb-8-zone` to the lisp handler.

**Comment by RetroDevil** [comment]
it's work when clevo touchpad selected but it's not smooth and stable like g502. It work like slow and laggy

**Comment by fraannk** [comment]
Tried the above build with the Rival 650, which is just a wireless version of the 600, and no luck. :(

**Comment by MithicalM** [comment]
@simon-wh I have had no luck with this build for the Rival 600. Mind if I ask if there is a more recent build that might be able to support it?

**Comment by MithicalM** [comment]
@simon-wh None of the builds so far have worked on my Steelseries Rival 600.

**Comment by MithicalM** [comment]
![image](https://user-images.githubusercontent.com/1751395/53675034-94233700-3c8a-11e9-890e-b5e61f0f0758.png)
this is what I get with both steel series mouse options.

**Comment by LittleThor** [comment]
@simon-wh Will you help us with this 
I tried all sorts of ways, but I couldn't reach a conclusion

**Comment by SnakePin** [comment]
Can you ppl test this build https://ci.appveyor.com/project/antonpup/aurora/builds/23481005/artifacts
Make sure to change layout to "SteelSeries - rgb8-zone - EXP"

**Comment by MithicalM** [comment]
![image](https://user-images.githubusercontent.com/1751395/55286116-27858000-538f-11e9-8e51-e25d61fd44ff.png)
Still no luck unfortunately...

**Comment by SnakePin** [comment]
Send your logs please and make sure steelseries integration is enabled and
running.

**Comment by MithicalM** [comment]
![image](https://user-images.githubusercontent.com/1751395/55292166-1c5a4080-53df-11e9-98ed-73dccbce2e1e.png)
I got it working now. I didn't realise that the Steel Series part of Device Manager was off. Turned it on and it works instantly. Thanks for getting it to work dude! Much appreciated!

**Comment by LittleThor** [comment]
It's working now but with this build rival 600 & any rgb mousepad (mine is steelseries qck prism xl) won't work together.

**Comment by RetroDevil** [comment]
It's working now but in this build key wave animation won't work.

**Comment by charmarkk** [comment]
Small update: Seems like this works for desktop wrapper support but breaks games. e.g. Overwatch wrappers and layers don't apply.

**Comment by 997R8V10** [comment]
Any update on this? Do I need to use the build listed here? The latest build doesn't seem to have 8 zone support.

**Comment by SnakePin** [comment]
@997R8V10 yep, you need to use the build I linked here https://github.com/antonpup/Aurora/issues/1199#issuecomment-478299297

**Comment by KarakNor** [comment]
So there are some bugs in this build, as described in this thread. Personally, I've experienced that key wave animations are not working in this build, although the mouse zones do work.

With beta v0.7.0rc2, basic support is there for this mouse, but only the logo and mouse wheel. Would be good to see a new build based on this, to see if the issues above can be addressed.
